app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,81 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all.order(:id)
+  end
+
+  def show
+    @task = Task.find_by(id: params[:id])
+    if @task.nil?
+      head :not_found
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    filtered_task_params = task_params()
+    @task = Task.new(filtered_task_params)
+    @task.completed = false
+    if @task.save
+      redirect_to tasks_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+  end
+
+  def update
+
+    task = Task.find_by(id: params[:id])
+
+    filtered_task_params = task_params()
+
+    if task.update(filtered_task_params)
+
+      redirect_to task_path(task)
+
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @task = Task.find_by(id: params[:id])
+    @task.destroy
+
+    redirect_to tasks_path
+
+  end
+
+  def complete
+    @task = Task.find(params[:id])
+    if @task.completed == false
+      @task.update(completed: true)
+      @task.update(completed_date: Date.today)
+    elsif @task.completed == true
```

Comment:
These two attributes are redundant! What should your code do if `completed` is `true`, but the `completion_date` is `nil`? Instead you might say that a `nil` value for `completion_date` indicates the task is incomplete, and a non-`nil` value indicates it is complete.